src/clients/Client.cc

```diff
@@ -455,33 +455,34 @@ sameUrlHosts(const char *url1, const char *url2)
 static void
 purgeEntriesByHeader(HttpRequest *req, const char *reqUrl, Http::Message *rep, Http::HdrType hdr)
 {
-    const char *hdrUrl, *absUrl;
-
-    absUrl = NULL;
-    hdrUrl = rep->header.getStr(hdr);
-    if (hdrUrl == NULL) {
+    const char *hdrUrl;
+    if (!(hdrUrl = rep->header.getStr(hdr)))
         return;
-    }
 
     /*
      * If the URL is relative, make it absolute so we can find it.
      * If it's absolute, make sure the host parts match to avoid DOS attacks
      * as per RFC 2616 13.10.
      */
+    SBuf absUrl;
```

Comment:
Let's clarify what is going on here -- this code is dangerous because keeping this seemingly excessively large absUrl scope is critical to code correctness.

```suggestion
    SBuf absUrlMaker;
    const char *absUrl = nullptr;
```

I have added one more explicit change request to sync with the above changes, but did not add the more obvious ones.